app/src/main/java/io/gnosis/safe/ui/splash/SplashViewModel.kt

```diff
@@ -12,13 +15,28 @@ import javax.inject.Inject
 class SplashViewModel
 @Inject
 constructor(
+    private val notificationRepository: NotificationRepository,
+    private val safeRepository: SafeRepository,
+    private val tracker: Tracker,
     private val termsChecker: TermsChecker,
     appDispatchers: AppDispatchers,
     @ApplicationContext private val appContext: Context
 ) : BaseStateViewModel<SplashViewModel.TermsAgreed>(appDispatchers) {
 
     override fun initialState(): TermsAgreed = TermsAgreed(null)
 
+    suspend fun onAppStart() {
+
+        notificationRepository.register()
+
+        notificationRepository.clearNotifications()
+        val pushEnabled = notificationRepository.checkPermissions()
+        tracker.setPushInfo(pushEnabled)
+
+        val numSafes = safeRepository.getSafes().size
```

Comment:
Should we encapsulate this in a method within the `SafeRepository` just in case at some point there is some magic involved on the safe list returned by `getSafes()` ?

Reply:
safeRepository.getNumberOfSafes() ?

Reply:
`getSafeCount` or what you said. Either option is fine :)